### PR TITLE
Allow integers for timeouts

### PR DESCRIPTION
**Pull request by traylenator**
Timeouts are more naturally written as integers, allow them.

We have run with integers here for many years.

### CODE REVIEW

**Comment by traylenator** [comment]
> [CLA assistant check](https://cla-assistant.io/puppetlabs/puppetlabs-apache?pullRequest=2294) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/puppetlabs/puppetlabs-apache?pullRequest=2294) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/puppetlabs/puppetlabs-apache?pullRequest=2294) it.

The CLA is signed on behalf of CERN IT.

File: manifests/mod/auth_cas.pp [hunk 1]
```diff
@@ -87,28 +87,28 @@
...
+  Stdlib::Filemode $cas_cookie_path_mode                 = '0750',
+  Integer $cas_version                                   = 2,
+  String $cas_debug                                      = 'Off',
+  Optional[String] $cas_validate_server                  = undef,
+  Optional[String] $cas_validate_depth                   = undef,
+  Optional[String] $cas_certificate_path                 = undef,
+  Optional[String] $cas_proxy_validate_url               = undef,
+  Optional[String] $cas_root_proxied_as                  = undef,
+  Optional[String] $cas_cookie_entropy                   = undef,
+  Optional[Variant[Integer[0],String]] $cas_timeout      = undef,
```
Comment (bastelfreak) [inline manifests/mod/auth_cas.pp:100]: could you add a unit test for the three params that set them to an integer? Is there an upper limit within apache that we could enforce with the type?

**Review by ekohl — COMMENTED** [review]
In ae56cc9a6dbcd3846ccbcb9f8a7b9302d8b2a996 initial data types were added and in f41251e336fa3e7c31c19968ccee74121cf71e47 we replaced a lot of the Strings with Integers, but clearly missed this. 

https://github.com/puppetlabs/puppetlabs-apache/pull/2290 is another instance of this. There I suggested to support a `Variant` (as you did here) and in https://github.com/puppetlabs/puppetlabs-apache/issues/2291 we're making notes to drop the compatibility in the next major version. It looks like your PR would supersede https://github.com/puppetlabs/puppetlabs-apache/pull/2290.

**Review by ekohl — APPROVED** [review]
I recall your org has signed the CLA, but the check is wrong. @david22swan @traylenator can you figure this out? The code looks good and solves a regression. After this a new release would be good to have.

**Review by david22swan — APPROVED** [review]
LGTM
Thank's for putting up the fix :)

As for the CLI, I'll take your word for it.
It's been buggy ever since it was implemented and this can happen.

**Pull request merged**